xmtp_id/src/scw_verifiers.rs

```diff
@@ -56,6 +65,29 @@ impl ERC1271Verifier {
 
         Ok(res == EIP1271_MAGIC_VALUE)
     }
+
+    /// Verifies an ERC-6492<https://eips.ethereum.org/EIPS/eip-6492> signature.
+    ///
+    /// # Arguments
+    ///
+    /// * `signer` - can be the smart wallet address or the ecrecover .
+    /// * `block_number` - Block number to verify the signature at.
+    /// * `hash` - Message digest for the signature.
+    /// * `signature` - Could be encoded smart wallet signature or raw ECDSA signature.
+    pub async fn erc6492_is_valid_sig(
+        &self,
+        signer: Address,
+        block_number: Option<BlockNumber>,
+        hash: [u8; 32],
+        signature: Bytes,
+    ) -> Result<bool, VerifierError> {
+        let erc6492 = ERC6492::new(signer, self.provider.clone());
+        Ok(erc6492
```

Comment:
Where does the universal verifier contract get referenced here? Don't we need that as a prerequisite to this call working?

In Viem, for example, it ephemerally deploys the verifier contract bytecode before calling the contract.

Reply:
I was using on chain validation where we expect a deployed `UniversalSigValidator` and call its `isValidSig` function.

Viem uses the off chain validation where it deploys `ValidateSigOffchain` contract that uses `UniversalSigValidator::isValidSig` logic in the constructor. 

Validation logic for both is identical but I figured we can only use on-chain solution given that off-chain consumes way more gas.

Reply:
Does gas matter? It reverts before any transaction goes through, so nothing actually gets spent.

The advantage of the offchain is that it works across all the chains, even ones where the validator contract hasn't been deployed yet.

Reply:
Thanks, I didn't know eth_call can do verification like that - I just made the changes per your suggestion.